src/main/java/algorithms/curated170/medium/ParallelCourses.java

```diff
@@ -0,0 +1,78 @@
+package algorithms.curated170.medium;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+public class ParallelCourses {
+    Map<Integer, HashSet<Integer>> courseChiMap;
+    Map<Integer, HashSet<Integer>> courseParMap;
+    HashSet<Integer> firstCourses = new HashSet<>();
+
+    public int solution(int[][] courses) {
+        createMap(courses);
+        return sortTopologically();
+    }
```

Comment:
Can we refactor the code to be compatible with LeetCode?

Default code is like this:
class Solution {
    public int minimumSemesters(int n, int[][] relations) {
        
    }
}